src/Find-SDC-Repositories.js

```diff
@@ -0,0 +1,119 @@
+const graphql = require("graphql-request");
+const { zipWith } = require("lodash");
+const prompt = require("prompt");
+const fs = require("fs");
+const process = require("process");
+const mkdirp = require("mkdirp");
+
+const config = {
+  file: {
+    filePrefix: "SDC-Teams-Github-Repos-",
+    directoryPath: "outputs/SDCTeamGithubRepos/"
+  },
+  api: {
+    endpoint: "https://api.github.com/graphql",
+    options: {
+      headers: {
+        Authorization: null
+      }
+    },
+    query: `
+    query { 
+        organization(login:"ONSDigital") { 
+            teams(first: 100) {
+                nodes {
+                    name
+                    repositories(first: 100) {
+                        nodes {
+                            name
+                        }
+                        edges {
+                            permission
+                        }
+                    }
+                }
+            }
+        }
+    }`
+  }
+};
+
+prompt.start();
+
+prompt.get(["key"], function(err, result) {
+  config.api.options.headers.Authorization = "Bearer " + result.key;
+
+  if (
+    config.api.options.headers.Authorization !== null &&
+    config.api.options.headers.Authorization !== ""
+  ) {
+    mkdirp(config.file.directoryPath);
+
+    const client = new graphql.GraphQLClient(
+      config.api.endpoint,
+      config.api.options
+    );
+
+    client
+      .request(config.api.query)
+      .then(function(data) {
+        let output = { teams: [] };
+
+        data.organization.teams.nodes.forEach(team => {
+          let teamData = {};
+
+          teamData.name = team.name;
+          teamData.repositories = [];
+
+          const names = team.repositories.nodes;
+          const permissions = team.repositories.edges;
+          const reposData = zipWith(
+            names,
+            permissions,
+            (names, permissions) => {
+              return {
+                name: names.name,
+                permission: permissions.permission
+              };
+            }
+          );
+
+          teamData.repositories.push(reposData);
+
+          output.teams.push(teamData);
+        });
+
+        output = JSON.stringify(output, null, 2);
+
+        let date = new Date();
+        fs.writeFile(
```

Comment:
Rather than writing to a file, I'd suggest it's probably worth just outputting to `stdout`. The user of the script can always pipe to a file themselves e.g. `script.js > output.json`

Reply:
All the logic below for constructing file name can be removed then, better to keep it simple 👍